File: service/src/main/java/no/nav/melosys/service/vedtak/FattVedtakDto.java
package no.nav.melosys.service.vedtak;

import no.nav.melosys.domain.kodeverk.Vedtakstyper;
import no.nav.melosys.domain.kodeverk.behandlinger.Behandlingsresultattyper;

public class FattVedtakDto {
    private final Behandlingsresultattyper behandlingsresultatTypeKode;
    private final Vedtakstyper vedtakstype;

    protected FattVedtakDto(Builder<?> builder) {
        this.behandlingsresultatTypeKode = builder.behandlingsresultatTypeKode;
        this.vedtakstype = builder.vedtakstype;
    }

    public Behandlingsresultattyper getBehandlingsresultatTypeKode() {
        return behandlingsresultatTypeKode;
    }

    public Vedtakstyper getVedtakstype() {
        return vedtakstype;
    }

    public abstract static class Builder<T extends Builder<T>> {
        private Behandlingsresultattyper behandlingsresultatTypeKode;
        private Vedtakstyper vedtakstype;

        public abstract T getThis();

        public T medBehandlingsresultat(Behandlingsresultattyper behandlingsresultatTypeKode) {
            this.behandlingsresultatTypeKode = behandlingsresultatTypeKode;
            return this.getThis();
        }

        public T medVedtakstype(Vedtakstyper vedtakstype) {
            this.vedtakstype = vedtakstype;
            return this.getThis();
        }

        public FattVedtakDto build() {
            return new FattVedtakDto(this);
        }
    }
}

File: frontend-api/src/main/kotlin/no/nav/melosys/tjenester/gui/dto/FattVedtakDto.kt
package no.nav.melosys.tjenester.gui.dto

import no.nav.melosys.domain.kodeverk.Vedtakstyper
import no.nav.melosys.domain.kodeverk.behandlinger.Behandlingsresultattyper
import no.nav.melosys.integrasjon.faktureringskomponenten.dto.FaktureringsIntervall
import no.nav.melosys.service.dokument.brev.KopiMottakerDto
import java.time.LocalDate

class FattVedtakDto {
    @JvmField
    var behandlingsresultatTypeKode: Behandlingsresultattyper? = null
    @JvmField
    var vedtakstype: Vedtakstyper? = null
    @JvmField
    var fritekst: String? = null
    @JvmField
    var fritekstSed: String? = null
    @JvmField
    var mottakerinstitusjoner: Set<String>? = null
    @JvmField
    var nyVurderingBakgrunn: String? = null
    @JvmField
    var innledningFritekst: String? = null
    @JvmField
    var begrunnelseFritekst: String? = null
    @JvmField
    var ektefelleFritekst: String? = null
    @JvmField
    var barnFritekst: String? = null
    @JvmField
    var trygdeavgiftFritekst: String? = null
    @JvmField
    var kopiMottakere: List<KopiMottakerDto>? = null
    @JvmField
    var betalingsintervall: FaktureringsIntervall? = null
    @JvmField
    var kopiTilArbeidsgiver: Boolean? = null
    @JvmField
    var opphoerDato: LocalDate? = null
}
